Bottom: 36fe9752394406c04c3d8f13fe21a0a3fdade499
Top:    a84f3a159dd9126f12ef6854a797b9e0dcdbf321
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-22 14:18:02 -0700

Refresh of dbg-fork

---

diff --git a/src/preload.c b/src/preload.c
index 0645f6d..cdca9df 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -413,11 +413,13 @@ int socket(int domain, int type, int protocol)
 		return index;
 
 	recursive = 1;
-	ret = rsocket(domain, type, protocol);
+//	ret = rsocket(domain, type, protocol);
+goto realsock;
 	recursive = 0;
 	if (ret >= 0) {
 		if (fork_support) {
 			rclose(ret);
+realsock:
 			ret = real.socket(domain, type, protocol);
 			if (ret < 0)
 				return ret;
